include/linux/double-fetch-detection.h

```diff
@@ -0,0 +1,17 @@
+#ifndef __LINUX_DOUBLE_FETCH_DETECTION_H__
+#define __LINUX_DOUBLE_FETCH_DETECTION_H__
+#ifdef CONFIG_DF_DETECTION
+
+struct address{
```

Comment:
If this is used elsewhere, you're going to prefix it with something to denote its "namespace".

Reply:
Can it be user_space_address or user_address_range? as it refers to addresses from userspace?

Reply:
I think it's still too general. I highly doubt anybody else will reuse this, because all you have here is a pointer+size, but otherwise nothing specific to addresses from userspace.

The usual convention is that for public identifiers across headers in include/linux is to prefix or somehow add the subsystem into each name (unless it's a core kernel primitive).

df_address_range might work, but I still don't like 'df' because many things could be 'df'.